src/FileSystem/DefaultFileSystem.cpp

```diff
@@ -72,6 +77,26 @@ std::filesystem::path DefaultFileSystem::FindPath(const std::filesystem::path& p
 	throw std::runtime_error("File " + path.string() + " not found");
 }
 
+bool DefaultFileSystem::IsPathValid(const std::filesystem::path& path)
+{
+	if (path.empty())
+	{
+		return false;
+	}
+
+	if (!Exists(path))
+	{
+		return false;
+	}
+
+	if (!std::filesystem::is_directory(path))
```

Comment:
There will be confusion on this function call if we rule out directories.
Either
1. Remove this restriction
2. Change the name so it's clear we're checking only directories
3. Add an option to the interface for the restriction